shm/player.cpp

```diff
@@ -0,0 +1,29 @@
+#include "player.hpp"
+
+Player::Player(std::shared_ptr<Ship> ship, size_t money, size_t availableSpace)
+    : ship_(ship)
+    , money_(money)
+    , availableSpace_(availableSpace)
```

Comment:
I think you should recalculate `availableSpace_` in palyer class. Each time you get cargo, you need to recalculate it again